test/sandbox/test/inline_erb_test.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class InlineErbTest < ViewComponent::TestCase
+  class InlineErbComponent < ViewComponent::Base
+    include ViewComponent::InlineTemplate
+
+    attr_reader :name
+
+    template! <<~ERB
+      <h1>Hello, <%= name %>!</h1>
+    ERB
+
+    def initialize(name)
+      @name = name
+    end
+  end
+
+  class InlineSlimComponent < ViewComponent::Base
+    include ViewComponent::InlineTemplate
+
+    attr_reader :name
+
+    template_language! :slim
```

Comment:
Why not make the language an argument of `template!`?

Reply:
I considered it, but I went with two different methods for a few reasons. The first is I wanted components to have a consistent templating language. e.g. if you have Slim and ERB in the same component that's going to be a bit difficult to read since you have to translate between the two.

The second reason is because I wanted `ApplicationComponent` to be able to set a default. That way all of your components can inherit the `template_language!` so it "just works™" instead of requiring callers to pass `slim` to each `template!` call.

Reply:
It would be awesome to be able to figure out the templating language based on the HEREDOC delimiter, but it does not seem possible, as a HEREDOC block is "just a string". I did not find a way to introspect the delimiter.

Reply:
@BlakeWilliams makes sense. Maybe we could set the language to be ERB by default?

Reply:
Sounds good, ERB is the default we fall back to if it's not already set: https://github.com/ViewComponent/view_component/pull/1556/files#diff-8678537e91c0b561c831072ad0aad8a3ed1208cf537591e96475a23959920b5dR27-R30